flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/PendingCheckpoint.java

```diff
@@ -130,7 +134,8 @@ public CompletedCheckpoint toCompletedCheckpoint() {
 	public boolean acknowledgeTask(
 			ExecutionAttemptID attemptID,
 			SerializedValue<StateHandle<?>> state,
-			long stateSize) {
+			long stateSize,
+			Map<Integer, SerializedValue<StateHandle<?>>> kvState) {
```

Comment:
The thing is that the `kvState` also needs the state size information. Thus, the method should get another parameter `Map<Integer, Long> kvStateSizes` and then I think the state size should be grouped together with the serialized state handle.